tools/lambda-promtail/lambda-promtail/s3.go

```diff
@@ -301,3 +332,16 @@ func stringToRawEvent(body string) (map[string]interface{}, error) {
 	}
 	return result, nil
 }
+
+func toMicroseconds(s string) (usec int64, err error) {
+	// Unix time in microseconds has 16 digits
+	i, err := strconv.ParseInt(s, 10, 64)
+	if err != nil {
+		return usec, err
+	}
+	iPow10 := int(math.Log10(float64(i)))
+	multiplier := math.Pow10(15 - iPow10)
+	usec = int64(float64(i) * multiplier)
+
+	return usec, err
+}
```

Comment:
could you add another comment in here about the transformation that's being done? an example would help